src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -167,7 +167,13 @@ private void ConfigureApplication(WebHostBuilderContext context, IApplicationBui
                 if (targetRouteBuilder is null)
                 {
                     // The app defined endpoints without calling UseRouting() explicitly, so call UseRouting() implicitly.
-                    app.UseRouting();
+                    app.UseRouting(overrideEndpointRouteBuilder: false);
+
+                    // Copy the endpoint route builder to the built application
```

Comment:
This is a bit hacky but required to ensure the EndpointRouteBuilder is shared between `_builtApplication` and `app`. We should probably invest in a cleaner approach for sharing the `Properties` bag between the two builders.